lib/kernel.coffee

```diff
@@ -496,3 +494,5 @@ class Kernel
 
         unless @onlyConnect
             @kernelProcess.kill 'SIGKILL'
+
+        @onDestroy(this)
```

Comment:
`onDestroy` should never be null, because that would be a resource leak. (It's needed to de-list the session from the ZMQGateway. A callback is the easiest way to avoid circular references)